diff --git a/node_modules/playwright-core/lib/utils/debugLogger.js b/node_modules/playwright-core/lib/utils/debugLogger.js
index e252f1c..d740d55 100644
--- a/node_modules/playwright-core/lib/utils/debugLogger.js
+++ b/node_modules/playwright-core/lib/utils/debugLogger.js
@@ -74,6 +74,12 @@ class DebugLogger {
       cachedDebugger.color = debugLoggerColorMap[name];
     }
 
+    if (name === "api") {
+      if (global.customLogger) {
+        global.customLogger.log(message);
+      }
+    }
+    
     cachedDebugger(message);
   }
 
