diff --git a/node_modules/shelljs/shell.js b/node_modules/shelljs/shell.js
index f9c6f36..f21b485 100644
--- a/node_modules/shelljs/shell.js
+++ b/node_modules/shelljs/shell.js
@@ -21,9 +21,33 @@ var common = require('./src/common');
 //@commands
 
 // Load all default commands
-require('./commands').forEach(function (command) {
-  require('./src/' + command);
-});
+require('./src/cat')
+require('./src/cd')
+require('./src/chmod')
+require('./src/cp')
+require('./src/dirs')
+require('./src/echo')
+require('./src/exec')
+require('./src/find')
+require('./src/grep')
+require('./src/head')
+require('./src/ln')
+require('./src/ls')
+require('./src/mkdir')
+require('./src/mv')
+require('./src/pwd')
+require('./src/rm')
+require('./src/sed')
+require('./src/set')
+require('./src/sort')
+require('./src/tail')
+require('./src/tempdir')
+require('./src/test')
+require('./src/to')
+require('./src/toEnd')
+require('./src/touch')
+require('./src/uniq')
+require('./src/which')
 
 //@
 //@ ### exit(code)
