diff --git a/shell.js b/shell.js
index f9c6f36df9de7c10958b01d08dbaa1b20034ad8c..f21b4851a51a6219a04249dd9dcb904ba96c375d 100644
--- a/shell.js
+++ b/shell.js
@@ -21,9 +21,33 @@ var common = require('./src/common');
 //@commands
 
 // Load all default commands
-require('./commands').forEach(function (command) {
-  require('./src/' + command);
-});
+require('./src/cat')
+require('./src/cd')
+require('./src/chmod')
+require('./src/cp')
+require('./src/dirs')
+require('./src/echo')
+require('./src/exec')
+require('./src/find')
+require('./src/grep')
+require('./src/head')
+require('./src/ln')
+require('./src/ls')
+require('./src/mkdir')
+require('./src/mv')
+require('./src/pwd')
+require('./src/rm')
+require('./src/sed')
+require('./src/set')
+require('./src/sort')
+require('./src/tail')
+require('./src/tempdir')
+require('./src/test')
+require('./src/to')
+require('./src/toEnd')
+require('./src/touch')
+require('./src/uniq')
+require('./src/which')
 
 //@
 //@ ### exit(code)
diff --git a/src/mkdir.js b/src/mkdir.js
index 6168d592b952292f25b40a2f1b5cbfd8f9e3b86c..8092db5e981b6519fda741d68bce5530be673632 100644
--- a/src/mkdir.js
+++ b/src/mkdir.js
@@ -19,17 +19,18 @@ function mkdirSyncRecursive(dir) {
     common.error('dirname() failed: [' + dir + ']');
   }
 
-  // Base dir exists, no recursion necessary
-  if (fs.existsSync(baseDir)) {
-    fs.mkdirSync(dir, parseInt('0777', 8));
-    return;
-  }
-
   // Base dir does not exist, go recursive
-  mkdirSyncRecursive(baseDir);
+  if (!fs.existsSync(baseDir)) {
+    mkdirSyncRecursive(baseDir);
+  }
 
-  // Base dir created, can create dir
-  fs.mkdirSync(dir, parseInt('0777', 8));
+  try {
+    // Base dir created, can create dir
+    fs.mkdirSync(dir, parseInt('0777', 8));
+  } catch (e) {
+    // swallow error if dir already exists
+    if (e.code !== 'EEXIST' || common.statNoFollowLinks(dir).isFile()) { throw e; }
+  }
 }
 
 //@